providers/default.rb

```diff
@@ -637,7 +637,9 @@ def execute_install_script(install_script)
                     '/etc/init.d/chef-client start'
                   end
       unless node['chef_client']['chef_license'].nil?
-        start_cmd = "env CHEF_LICENSE=\"#{node['chef_client']['chef_license']}\" " + start_cmd
+        env = {}
+        env['CHEF_LICENSE'] = node['chef_client']['chef_license']
+        fork { Kernel.exec(env, "#{chef_install_dir}/bin/chef-apply", '-e "exit 0"') }
```

Comment:
so why aren't we using `shell_out` here?  i'm guessing that the prior attempt to use `chef-client -z` failed to exit due to the sub-chef-client spinning on the lockfile, which led to this kind of fire-and-forget attempted solution, but that now seems wrong.

obliterating the whole env other than CHEF_LICENSE also seems like its likely to have issues.  that'll remove the PATH and HOME and bunch of other env vars that may be necessary.

Reply:
it should probably use `shell_out` with a reasonable `timeout` value like 60 seconds or so, then take the exitcode and if the command fails Chef::Log.warn the user that something went wrong with the attempt at license acceptance.